cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/xcm_config.rs

```diff
@@ -55,6 +61,7 @@ parameter_types! {
 		X2(GlobalConsensus(RelayNetwork::get()), Parachain(ParachainInfo::parachain_id().into()));
 	pub const MaxInstructions: u32 = 100;
 	pub const MaxAssetsIntoHolding: u32 = 64;
+	pub TreasuryAccount: Option<AccountId> = Some(TREASURY_PALLET_ID.into_account_truncating());
```

Comment:
Having a single global treasury account to hold collected fees for all bridges may be problematic, from an operational perspective.

Our Ethereum bridge collects XCM delivery fees for `ExportMessage`, and then uses those funds to pay incentivize message relayers.

For example, If our bridge (or any other one) is misconfigured, it could potentially drain `TreasuryAccount` empty, and potentially break other bridges who are also holding funds in that account.

Another reason to have separate accounts is to know at a glance how much funds each bridge has collected. As part of our bridge operations, we need to periodically withdraw DOT from the treasury, exchange it for ETH and deposit that ETH on the Ethereum side of the bridge.

We could implement multiple treasury accounts by modifying `ExportXcm::validate` function so that it returns some kind of ID or address which can be used by the XCM executor in conjunction with `FeeManager` to select a destination account for fees.

@acatangiu thoughts?

Reply:
That's a good point @vgeddes, thanks for spotting this!

Right now BH is charging "bridge fee" as part of `ExportMessage` XCM **execution** fee. So, if we stay with this model we can't afaict charge different execution costs based on destination (execution cost is instruction weight), and we'll end up doing `Max(to-Kusama, to-Ethereum)` and maybe return "unspent weight" from exporter.

We should also explore using/or switching over to the _XCM delivery fees_ system added by this PR and see if how to best charge differently based on exporter/destination.

Either way, I agree with you that we ideally want separate accounts between bridges, but since this is SP owned by governance maybe these would be some treasury sub-accounts? @joepetrowski wdyt can/should we do this?

Reply:
@KiChjang regardless of how we design/split bridges "transport fee"s between XCM delivery and XCM execution, I think @vgeddes raised a good point on the general "design" of this `XcmFeesToAccount fee manager`. For other usecases too, it should be configurable on both:
- fee value based on destination
- target account for fee deposit based on destination

WDYT?

Reply:
The `FeeManager` is not responsible for calculating the fees required for delivery or execution; you'll see that in the fn sig of `handle_fees`, it accepts a `fees` parameter, i.e. another component is responsible for taking the fees required from a payer account, and then providing it to the `FeeManager`.

All `FeeManager` really does is manage:

1.  whether the fees can be waived or not;
2. what we should do with the `fees` collected.

So your first point is inapplicable, whereas on the 2nd point, we can already do so via the `ReceiverAccount` type parameter, albeit not based on the destination -- supporting this feature requires a bit more thought in redesigning the API to include destination info.

Reply:
thanks for the explanation!

> whereas on the 2nd point, we can already do so via the ReceiverAccount type parameter, albeit not based on the destination -- supporting this feature requires a bit more thought in redesigning the API to include destination info

If you don't plan on tackling this in current PR (don't want to block on this, I think we can also add it afterwards), can you please open an issue describing your view of it?

Reply:
I would tackle this in another PR so we can merge this.
I created an issue for it: https://github.com/paritytech/polkadot-sdk/issues/1549

Reply:
@vgeddes 
Published https://github.com/paritytech/polkadot-sdk/pull/2021 for this. Can you please take a look ?